### PR TITLE
[Stream Analytics] Fix S360 for auto-scale feature

### PR DESCRIPTION
**Pull request by yanlingms**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [x] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
ASAP.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
ASAP.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [x] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/preview/2021-10-01-preview/streamingjobs.json [hunk 1]
```diff
@@ -571,6 +571,51 @@
         },
         "x-ms-long-running-operation": true
       }
+    },
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.StreamAnalytics/streamingjobs/{jobName}/skus": {
```
Comment (raosuhas) [inline specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/preview/2021-10-01-preview/streamingjobs.json:575]: >skus [](http://example.com/codeflow?start=132&length=4)

this API needs to return a list of ARM objects and support individual gets on each object. 


i.e : 
GET .../streamingjobs/{jobname}/skus/{skuname}


Your other option if you only want to return a non-ARM formatted object is to change this into a POST method s.. i.e something like : 
POST .../streamingjobs/{jobname}/listskus

File: specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/preview/2021-10-01-preview/streamingjobs.json [hunk 2]
```diff
@@ -571,6 +571,51 @@
...
+        "operationId": "StreamingJobs_GetSkus",
+        "description": "Gets a list of available SKUs about the specified streaming job.",
+        "x-ms-examples": {
+          "Get valid SKUs list for the specified streaming job": {
+            "$ref": "./examples/StreamingJob_GetSkus.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../common/v1/definitions.json#/parameters/ApiVersionParameter"
```
Comment (raosuhas) [inline specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/preview/2021-10-01-preview/streamingjobs.json:589]: >v1 [](http://example.com/codeflow?start=34&length=2)

consider using v3 definitions

File: specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/preview/2021-10-01-preview/examples/StreamingJob_GetSkus.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{
+  "parameters": {
+    "api-version": "2022-03-01-preview",
```
Comment (authuir) [inline specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/preview/2021-10-01-preview/examples/StreamingJob_GetSkus.json:3]: Supposed to be 2021-10-01-preview.

File: specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/preview/2021-10-01-preview/streamingjobs.json (hunk 1 above)
Comment (authuir) [reply specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/preview/2021-10-01-preview/streamingjobs.json:575]: I think the GET SKUs API is supposed to be list only, here is an example on VM get sku api:

https://docs.microsoft.com/en-us/rest/api/compute/resource-skus/list?tabs=HTTP

File: specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/preview/2021-10-01-preview/streamingjobs.json (hunk 2 above)
Comment (yanlingms) [reply specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/preview/2021-10-01-preview/streamingjobs.json]: Will this definition changes affect current behavior?

Comment (yanlingms) [reply specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/preview/2021-10-01-preview/streamingjobs.json:589]: We have no v3 definitions in path "../../common/", this file is defined by ourselves.

Comment (raosuhas) [reply specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/preview/2021-10-01-preview/streamingjobs.json:589]: oh k sorry i misread as common types

File: specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/preview/2021-10-01-preview/streamingjobs.json (hunk 1 above)
Comment (raosuhas) [reply specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/preview/2021-10-01-preview/streamingjobs.json:575]: Sorry about the confusion, this is correct. Approving the PR now.

**Comment by tadelesh** [comment]
@yanlingms Could you help to address the lint diff error and swagger breaking changes?

**Comment by yanlingms** [comment]
Hey @tadelesh, I've fixed the lint diff error, but how could I fix the ones in swagger breaking changes?

File: specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/preview/2021-10-01-preview/streamingjobs.json (hunk 1 above)
Comment (tadelesh) [inline specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/preview/2021-10-01-preview/streamingjobs.json:575]:
```suggestion
    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.StreamAnalytics/streamingjobs/{jobName}/skus": {
```

**Comment by tadelesh** [comment]
> Hey @tadelesh, I've fixed the lint diff error, but how could I fix the ones in swagger breaking changes?

You need to follow [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes) to get breaking change review.

File: specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/preview/2021-10-01-preview/streamingjobs.json (hunk 1 above)
Comment (yanlingms) [reply specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/preview/2021-10-01-preview/streamingjobs.json:575]: I guess we'd better to follow the other API's format that use `resourcegroups`.

**Comment by yanlingms** [comment]
> > Hey @tadelesh, I've fixed the lint diff error, but how could I fix the ones in swagger breaking changes?
> 
> You need to follow [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes) to get breaking change review.

Does adding new properties or new APIs also need breaking change review?

**Comment by tadelesh** [comment]
> > > Hey @tadelesh, I've fixed the lint diff error, but how could I fix the ones in swagger breaking changes?
> > 
> > 
> > You need to follow [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes) to get breaking change review.
> 
> Does adding new properties or new APIs also need breaking change review?

You add them in the current api version and this need to get reviewed.

**Comment by yanlingms** [comment]
Hey @tadelesh, the breaking changes have been approved, could we merge this PR?

**Review by tadelesh — APPROVED** [review]

**Pull request merged**